### PR TITLE
Proposed Process update

### PR DESCRIPTION
**Pull request by erica**
I found the original process.md document hard to follow. Here's a proposal for a more readable version.

### CODE REVIEW

**Comment by lattner** [comment]
Please remove the process-original.md addition.  @DougGregor should review this.

**Comment by erica** [comment]
I *think* I did it right but I'm a complete moron when it comes to git (which is why all my repos have shell scripts). Can you please check and let me know if I did it or I need to do something more.

Thanks, -- E

**Comment by jessesquires** [comment]
@erica :laughing: Looks good, but you'll probably need to squash the 2 commits into 1.

**Comment by jackhl** [comment]
It might be worth having a quick DevPubs/editorial pass on this document.

**Comment by erica** [comment]
Probably a good idea since I already see a few typos:

Is it an evolutionary change to the Swift language or Swift standard library public interface?
Is it a new language features or APIs? (no matter how small!)
Does it change existing language features and APIs? 
Does it removal of existing features from the language?
should read

Is it an evolutionary change to the Swift language or to Swift standard library public interface?
Is it a new language feature or API? (no matter how small!)
Does it change any existing language features or APIs? 
Does it propose removing existing features from the language?
-- E, who is very bad at (1) self-copy edits, and (2) writing in markdown

**Comment by erica** [comment]
I honestly have no idea how that is done. Any tips would be welcome.

-- E

**Comment by Danappelxx** [comment]
@erica You should read this, http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html but in essence:

1. `git rebase -i HEAD~2` - rebase the last 2 commits
2. For the commit which you want to merge, change "pick" to "squash"
3. Save & exit

**Comment by erica** [comment]
I don't think I can be trusted with git.

**Comment by erica** [comment]
I have deleted my fork. And re-submitted a separate pull request.

**Comment by lattner** [comment]
> On Dec 6, 2015, at 3:49 PM, Erica Sadun <notifications@github.com> wrote:
> 
> I *think* I did it right but I'm a complete moron when it comes to git (which is why all my repos have shell scripts). 

FWIW, I have a hard time believing that anyone is a bigger git-moron than me.  :-)

-Chris

**Comment by mcdado** [comment]
@erica allow me to humbly suggest that if you're not just doing everything via Github web interface, you could give [GitUp](http://gitup.co) a try, instead of the confusing command line git commands. GitUp gives you a very visual way to work with your local copy of the repository.

It uses some idioms and enables some actions that are not native to git, but they make a lot of sense:
- *rewriting commits*: technically checking out a former commit and amending to it, then rebasing the following commits onto it. it enables you to correct a mistake or make changes to a previous commit.
- *edit commit messages*: as simple as that sounds, edit the message of a previous commit
- etc.

I need to warn though that this causes what is referred to as changing history, and it has nasty consequences if the commit in question is already merged or somebody else is basing work on top of it. As long as you're always creating a new separate branch on your fork for each single proposed change this should not be a problem (if you propose a change via Github, this is usually done automatically with `patch-x` branch names). You just need to force push to your own forked branch after performing one of the actions above.

After you start proposing a change via Github web interface, if you add your fork as a remote to your locally cloned repo, you'll be able to pull and push to the newly created branch by Github. So when somebody asks to rebase or change something, you don't need to create a new branch or redo the work: just make the changes locally and force push.

I hope I made myself clear enough, I'm sorry if I didn't 😉
